### PR TITLE
Generate a list of supported DOM APIs from parsed WebIDLs.

### PR DESCRIPTION
**Pull request by jdm**
Supplement the existing list of supported CSS properties with an equivalent list of DOM APIs.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because automatic generated documentation

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12451)

### CODE REVIEW

File: etc/ci/upload_docs.sh [hunk 1]
```diff
@@ -16,5 +16,7 @@ cp etc/doc.servo.org/* target/doc/
 
 python components/style/properties/build.py servo html
 
+OUT_DIR=`pwd`/target/doc make -f makefile.cargo -C components/script dom_docs
```
Comment (aneeshusa) [inline etc/ci/upload_docs.sh]: Please use `OUT_DIR="$(pwd)/target/doc"` per the [style guide](https://github.com/servo/servo/blob/master/docs/STYLE_GUIDE.md).

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 2]
```diff
@@ -6437,3 +6437,36 @@ def UnionTypes(config):
 
         # Done.
         return curr
+
+    @staticmethod
+    def SupportedDomApis(config):
+        import os
```
Comment (aneeshusa) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Why not import this at the top level?

**Comment by jdm** [comment]
Sample output: http://www.joshmatthews.net/apis.html

**Comment by Ms2ger** [comment]
We might want to show something nicer than `__namedgetter()`; looks fine otherwise.

**Comment by jdm** [comment]
I hid every method that doesn't have an identifier for the time being (ie. __stringifier, __namedgetter, etc.)

**Pull request merged**